syntax = "proto2";
package protocol.pipe;

message pmsg_req_login {
	required string uid = 1;
	required string token = 2;
	required string pt = 3;
}

message pmsg_rep_login {
	required int32 errres = 1;
	required string errmsg = 2;
}

message pmsg_gonggao {
	required string gonggao = 1;
}

message pmsg_req_libao {
	required string code = 1;
	required string use = 2;
}

message pmsg_rep_libao {
	required int32 res = 1;
	required int32 pc = 2;
	required int32 cf = 3;
	repeated int32 types = 4;
	repeated int32 value1 = 5;
	repeated int32 value2 = 6;
	repeated int32 value3 = 7;
}

message pmsg_mail_one {
	repeated string acc_names		= 1;
	required int32 serverid		= 2;
	required string	title			= 4;	// 邮件标题
	required string	text			= 5;	// 邮件正文
	required string	sender_name		= 6;	// 发件人
	repeated int32 type				= 7;
	repeated int32 value1			= 8;
	repeated int32 value2			= 9;
	repeated int32 value3			= 10;
}

message pmsg_mail_all {
	required int32 level			= 1;
	required int32 serverid		= 2;
	required string	title			= 4;	// 邮件标题
	required string	text			= 5;	// 邮件正文
	required string	sender_name		= 6;	// 发件人
	repeated int32 type				= 7;
	repeated int32 value1			= 8;
	repeated int32 value2			= 9;
	repeated int32 value3			= 10;
}

message pmsg_req_recharge{
	required string pt				= 1;
	repeated string code			= 2;
}

message pmsg_recharge_ali {
	required uint64 guid			= 1;
	required int32 rid				= 2;
	required string orderno			= 3;
	required float amount			= 4;
}

message pmsg_rep_recharge{
	required int32 res				= 1;
	required string product_id		= 2;
	required string orderid			= 3;
}

message pmsg_recharge_simulation {
	required string acc_names		= 1;
	required int32 serverid			= 2;
	required int32 rid				= 3;
}

message pmsg_recharge_simulation1 {
	required uint64 guid			= 1;
	required int32 rid				= 2;
}

message pmsg_rank_forbidden {
	required string acc_names		= 1;
	required int32 serverid			= 2;
}

message pmsg_rank_forbidden1 {
	required uint64 guid			= 1;
}